File: app/src/main/java/com/chooloo/www/callmanager/ui/fragment/cursor/contacts/ContactsContract.java
package com.chooloo.www.callmanager.ui.fragment.cursor.contacts;

import com.chooloo.www.callmanager.database.entity.Contact;
import com.chooloo.www.callmanager.ui.fragment.cursor.CursorContract;

public interface ContactsContract extends CursorContract {
    interface View extends CursorContract.View {
        void openContact(Contact contact);

        String getHeader(int position);

        void showAnchoredHeader(boolean isShow);

        void setAnchoredHeader(String header);

        void refreshHeaders();

        void updateFastScrollerPosition();

        int getFirstVisibleItem();

        int getFirstCompletelyVisibleItem();
    }

    interface Presenter<V extends View> extends CursorContract.Presenter<V> {
        void onContactItemClick(Contact contact);

        void onContactItemLongClick(Contact contact);
    }
}

File: app/src/main/java/com/chooloo/www/koler/ui/contacts/ContactsContract.kt
package com.chooloo.www.koler.ui.contacts

import com.chooloo.www.koler.data.account.ContactAccount
import com.chooloo.www.koler.ui.list.ListContract

interface ContactsContract : ListContract {
    interface View : ListContract.View<ContactAccount> {
        fun openContact(contact:ContactAccount)
    }

    interface Controller<V : View> : ListContract.Controller<ContactAccount, V> {
    }
}